src/webgpu/shader/execution/evaluation_order.spec.ts

```diff
@@ -0,0 +1,480 @@
+      _result: 20 - 20,
+    },
+    {
+      name: 'RightSE', //
+      _body: 'return a - mul(&a, 10);',
+      _result: 2 - 20,
+    },
+    {
+      name: 'ThreeSE',
+      _body: 'return mul(&a, 10) - mul(&a, 10) - mul(&a, 10);',
+      _result: 20 - 200 - 2000,
+    },
+    {
+      name: 'LeftmostSE',
+      _body: 'return mul3_ret0(&a, &b, &c, 10) - a - b - c;',
+      _result: 0 - 20 - 30 - 40,
+    },
+    {
+      name: 'RightmostSE', //
+      _body: 'return a - b - c - mul3_ret0(&a, &b, &c, 10);',
+      _result: 2 - 3 - 4 - 0,
+    },
+    {
+      name: 'MiddleSE', //
+      _body: 'return a - b - mul3_ret0(&a, &b, &c, 10) - c;',
+      _result: 2 - 3 - 0 - 40,
+    },
+    {
+      name: 'ConstAndSEAndVar', //
+      _body: 'return 1 - mul(&a, 10) - a;',
+      _result: 1 - 20 - 20,
+    },
+    {
+      name: 'VarAndSEAndConst', //
+      _body: 'return a - mul(&a, 10) - 1;',
+      _result: 2 - 20 - 1,
+    },
+    {
+      name: 'SEAndVarAndConst', //
+      _body: 'return mul(&a, 10) - a - 1;',
+      _result: 20 - 20 - 1,
+    },
+    {
+      name: 'VarAndConstAndSE', //
+      _body: 'return a - 1 - mul(&a, 10);',
+      _result: 2 - 1 - 20,
+    },
+    {
+      name: 'MemberAccessAndSE',
+      _body: 'return vec4_zero.x + set_vec4_x(&vec4_zero, 42);',
+      _result: 0,
+    },
+    {
+      name: 'SEAndMemberAccess',
+      _body: 'return set_vec4_x(&vec4_zero, 42) + vec4_zero.x;',
+      _result: 42,
+    },
+  ])
+  .fn(t => run(t, t.params._body, t.params._result));
+
+g.test('binary_logical')
+  .specURL('https://gpuweb.github.io/gpuweb/wgsl/#logical-expr')
+  .desc('Tests order of evaluation of logical binary expressions.')
+  .paramsSimple([
+    {
+      name: 'BothSE',
+      _body:
+        'let r = (add(&a, 1) == 3) && (add(&a, 1) == 4);' + //
+        'return bool_to_i32(r);',
+      _result: 1,
+    },
+    {
+      name: 'LeftSE',
+      _body:
+        'let r = (add(&a, 1) == 3) && (a == 3);' + //
+        'return bool_to_i32(r);',
+      _result: 1,
+    },
+    {
+      name: 'RightSE',
+      _body:
+        'let r = (a == 2) && (add(&a, 1) == 3);' + //
+        'return bool_to_i32(r);',
+      _result: 1,
+    },
+    {
+      name: 'LeftmostSE',
+      _body:
+        'let r = (mul3_ret0(&a, &b, &c, 10) == 0) && (a == 20) && (b == 30) && (c == 40);' +
+        'return bool_to_i32(r);',
+      _result: 1,
+    },
+    {
+      name: 'RightmostSE',
+      _body:
+        'let r = (a == 2) && (b == 3) && (c == 4) && (mul3_ret0(&a, &b, &c, 10) == 0);' +
+        'return bool_to_i32(r);',
+      _result: 1,
+    },
+    {
+      name: 'MiddleSE',
+      _body:
+        'let r = (a == 2) && (b == 3) && (mul3_ret0(&a, &b, &c, 10) == 0) && (c == 40);' +
+        'return bool_to_i32(r);',
+      _result: 1,
+    },
+    {
+      name: 'ShortCircuit_And_LhsOnly',
+      _body:
+        // rhs should not execute
+        'let t = (a != 2) && (mul(&a, 10) == 20);' + //
+        'let r = (a == 2);' +
```

Comment:
You could just return `a` here and reduce the noise a bit. Here and below.

Reply:
Done.